spec/docs/expressions.md

```diff
@@ -0,0 +1,62 @@
+# Generating values with expressions
+
+<dfn>Expressions</dfn> are mapping constructs that can be evaluated on a [=logical iteration=], according to the specified reference formulation, to generate values during the mapping process.
+
+## Expression map (`rml:ExpressionMap`)
+
+An <dfn>expression map</dfn> (`rml:ExpressionMap`) is an abstract class, that is specialized by other RML classes. An [=expression map=] MAY have one the following properties:
+* 0 or 1 `rml:constant`, or
+* 0 or 1 `rml:reference`, or
+* 0 or 1 `rml:template`.
+
+Each of these properties specifies an [=expression=] which, upon evaluation, results in a set of values.
+
+The <dfn>reference expression set</dfn> of an [=expression map=] ar the set of expressions which are evaluated on a [=logical iteration=].
+
+### Constant expression (`rml:constant`)
+
+A <dfn>constant-valued expression map</dfn> is an [=expression map=] that always generates the same expression value. A constant-valued expression map is represented by a resource that has exactly one `rml:constant` property, the value of which is called a <dfn>constant expression</dfn>.
+
+The <dfn>constant value</dfn> is a singleton set containing the [=constant expression=].
+
+The [=reference expressions=] of a [constant-valued expression map=] is the empty set.
+
+### Reference (`rml:reference`)
+A <dfn>reference-valued expression map</dfn> is an [=expression map=] that is represented by a resource that has exactly one `rml:reference` property, the value of which is called a <dfn>reference expression</dfn>.
```

Comment:
We should define what should happen if a parameter is missing from the template. I would say that this returns `null` (or something like that).

Reply:
Do you refer here to the reference or to the template?

Reply:
Well if a template has 3 parameters, one of them is missing. What should the Processor do?
IMO: return `null` to skip it. However, we should define it because you can also have 'best effort' where you would just output the template with only 2 out of 3 parameters filled in.